File: java_spring-boot_spring-webflux_spring-data-jpa_lombok/src/main/java/org/example/demo/entity/Employee.java
package org.example.demo.entity;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Entity
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String firstname;
    private String lastname;
    private LocalDate dateOfBirth;

    public Employee update(Employee employee) {
        return Employee.builder()
                .id(employee.getId())
                .dateOfBirth(employee.dateOfBirth)
                .firstname(employee.getFirstname())
                .lastname(employee.getLastname())
                .build();
    }
}

File: kotlin_spring-boot_spring-boot-data-jpa_spring-web_maven/src/main/kotlin/org/example/demoapp/employee/Employee.kt
package org.example.demoapp.employee

import java.time.LocalDate
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
data class Employee(
        @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,
        var firstname: String? = null,
        var lastname: String? = null,
        var dateOfBirth: LocalDate? = null) {

    fun updateFrom(updatedEmployee: Employee) {
        this.firstname = updatedEmployee.firstname
        this.lastname = updatedEmployee.lastname
        this.dateOfBirth = updatedEmployee.dateOfBirth
    }
}